### PR TITLE
[Scripts] Doll Updates

### PR DESCRIPTION
**Pull request by imgntn**
This updates the doll to:
- Stop making noise when you let go
- Only respect input from the initial grab hand
- Spatialize audio
- Smoother animations

Create a doll by using the following script.  Also please use the following grab script.
https://rawgit.com/imgntn/hifi/doll_updates_1/examples/toys/doll/createDoll.js
https://rawgit.com/ericrius1/hifi/381a24951e68ccff1b4ce52bdd350cd20be70dea/examples/controllers/handControllerGrab.js

### CODE REVIEW

File: examples/toys/doll/createDoll.js [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+/*global MyAvatar, Entities, AnimationCache, SoundCache, Scene, Camera, Overlays, Audio, HMD, AvatarList, AvatarManager, Controller, UndoStack, Window, Account, GlobalServices, Script, ScriptDiscoveryService, LODManager, Menu, Vec3, Quat, AudioDevice, Paths, Clipboard, Settings, XMLHttpRequest, randFloat, randInt */
+
+function createDoll() {
+    var modelURL = "http://hifi-public.s3.amazonaws.com/models/Bboys/bboy2/bboy2.fbx";
+    var scriptURL = Script.resolvePath("doll.js");
+    var center = Vec3.sum(Vec3.sum(MyAvatar.position, {
+        x: 0,
+        y: 0.5,
+        z: 0
+    }), Vec3.multiply(0.5, Quat.getFront(Camera.getOrientation())));
```
Comment (ZappoMan) [inline examples/toys/doll/createDoll.js]: even based on the most recent coding standard conversations, this wouldn't be allowed by the new proposed standard. please fix this. please change your "beautifier" to follow the coding standard or stop using it.

File: examples/toys/doll/createDoll.js [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+    var desiredDimensions = Vec3.multiply(naturalDimensions, 0.15);
+    var doll = Entities.addEntity({
+        type: "Model",
+        name: "doll",
+        modelURL: modelURL,
+        script: scriptURL,
+        position: center,
+        shapeType: 'box',
+        dimensions: desiredDimensions,
+        gravity: {
```
Comment (ZappoMan) [inline examples/toys/doll/createDoll.js]: excessive verticalization

File: examples/toys/doll/createDoll.js [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+            x: 0,
+            y: 0,
+            z: 0
+        },
+        velocity: {
```
Comment (ZappoMan) [inline examples/toys/doll/createDoll.js]: excessive verticalization

File: examples/toys/doll/doll.js [hunk 4]
```diff
@@ -0,0 +1,92 @@
...
+//  Created by Eric Levin on 9/21/15.
+//  Copyright 2015 High Fidelity, Inc.
+//
+//  ATTENTION: requires that you run handController.js
+//  This entity script plays an animation and a sound while you hold it.
+//  Distributed under the Apache License, Version 2.0.
+//  See the accompanying file LICENSE or http://www.apache.org/licenses/LICENSE-2.0.html
+//
+/*global MyAvatar, Entities, AnimationCache, SoundCache, Scene, Camera, Overlays, Audio, HMD, AvatarList, AvatarManager, Controller, UndoStack, Window, Account, GlobalServices, Script, ScriptDiscoveryService, LODManager, Menu, Vec3, Quat, AudioDevice, Paths, Clipboard, Settings, XMLHttpRequest, randFloat, randInt */
+(function () {
```
Comment (ZappoMan) [inline examples/toys/doll/doll.js]: no space before ()

File: examples/toys/doll/doll.js [hunk 5]
```diff
@@ -0,0 +1,92 @@
...
+    Script.include("../../utilities.js");
+    Script.include("../../libraries/utils.js");
+
+    // this is the "constructor" for the entity as a JS object we don't do much here
+    var Doll = function () {
```
Comment (ZappoMan) [inline examples/toys/doll/doll.js]: no space before ()

File: examples/toys/doll/doll.js [hunk 6]
```diff
@@ -0,0 +1,92 @@
...
+            startFrame: 0,
+            lastFrame: 128,
+            startAutomatically: true
+        }),
+        stopAnimationSetting: JSON.stringify({
+            running: false,
+        }),
+        audioInjector: null,
+        isGrabbed: false,
+        setLeftHand: function () {
```
Comment (ZappoMan) [inline examples/toys/doll/doll.js]: no space before ()

File: examples/toys/doll/doll.js [hunk 7]
```diff
@@ -0,0 +1,92 @@
...
+            this.hand = 'left';
+        },
+
+        setRightHand: function () {
+            this.hand = 'right';
+        },
+
+        startNearGrab: function () {
```
Comment (ZappoMan) [inline examples/toys/doll/doll.js]: no space before ()

File: examples/toys/doll/doll.js [hunk 8]
```diff
@@ -0,0 +1,92 @@
...
+                    position: position,
+                    volume: 0.1
+                });
+                this.isGrabbed = true;
+                this.initialHand = this.hand;
+
+            }
+        },
+
+        continueNearGrab: function () {
```
Comment (ZappoMan) [inline examples/toys/doll/doll.js]: no space before ()

File: examples/toys/doll/doll.js [hunk 9]
```diff
@@ -0,0 +1,92 @@
...
+            var position = Entities.getEntityProperties(this.entityID, "position").position;
+            var audioOptions = {
+                position: position
+            };
+            this.audioInjector.options = audioOptions;
+        },
+
+        releaseGrab: function () {
```
Comment (ZappoMan) [inline examples/toys/doll/doll.js]: no space before ()

File: examples/toys/doll/doll.js [hunk 10]
```diff
@@ -0,0 +1,92 @@
...
+                    animationURL: "http://hifi-public.s3.amazonaws.com/models/Bboys/bboy2/bboy2.fbx",
+                });
+                this.isGrabbed = false;
+            }
+        },
+
+        // preload() will be called when the entity has become visible (or known) to the interface
+        // it gives us a chance to set our local JavaScript object up. In this case it means:
+        //   * remembering our entityID, so we can access it in cases where we're called without an entityID
+        preload: function (entityID) {
```
Comment (ZappoMan) [inline examples/toys/doll/doll.js]: no space before ()

File: examples/toys/doll/doll.js [hunk 11]
```diff
@@ -0,0 +1,92 @@
...
+        stopAnimationSetting: JSON.stringify({
+            running: false,
+        }),
+        audioInjector: null,
+        isGrabbed: false,
+        setLeftHand: function () {
+            this.hand = 'left';
+        },
+
+        setRightHand: function () {
```
Comment (ZappoMan) [inline examples/toys/doll/doll.js]: no space before ()

File: examples/toys/doll/doll.js [hunk 12]
```diff
@@ -0,0 +1,92 @@
...
+            var audioOptions = {
+                position: position
+            };
+            this.audioInjector.options = audioOptions;
+        },
+
+        releaseGrab: function () {
+            if (this.isGrabbed === true && this.hand === this.initialHand) {
+                this.audioInjector.stop();
+                Entities.editEntity(this.entityID, {
```
Comment (ZappoMan) [inline examples/toys/doll/doll.js]: DO NOT call Entities.editEntity(this.entityID) with the same entityID multiple times... make one call.

```
Entities.editEntity(this.entityID, {
    animationSettings: this.stopAnimationSetting,
    animationURL: "http://hifi-public.s3.amazonaws.com/models/Bboys/bboy2/bboy2.fbx",
});
```

**Comment by imgntn** [comment]
Bump for review :)

File: examples/toys/doll/doll.js [hunk 13]
```diff
@@ -0,0 +1,93 @@
...
+            };
+            this.audioInjector.options = audioOptions;
+        },
+
+        releaseGrab: function() {
+                if (this.isGrabbed === true && this.hand === this.initialHand) {
+
+                    this.audioInjector.stop();
+
+                Entities.editEntity(this.entityID, {
```
Comment (ZappoMan) [inline examples/toys/doll/doll.js]: There appears to be bad indenting here. This line should lineup with the statement above it. Is it possible you have tabs?

**Pull request merged**